code/game/objects/items/implants/implantpad.dm

```diff
@@ -59,12 +62,19 @@
 
 /obj/item/implantpad/ui_static_data(mob/user)
 	var/list/data = list()
-	data["has_case"] = !!inserted_case
+	data["has_case"] = inserted_case ? TRUE : FALSE
 	if(!inserted_case)
 		return data
-	data["has_implant"] = !!inserted_case.imp
+	data["has_implant"] = inserted_case.imp ? TRUE : FALSE
 	if(inserted_case.imp)
 		data["case_information"] = inserted_case.imp.get_data()
+		data["case_lore"] = inserted_case.imp.get_lore()
+		if(istype(inserted_case.imp, /obj/item/implant/deathrattle))
+			var/obj/item/implant/deathrattle/inserted_deathrattle = inserted_case.imp
+			if(inserted_deathrattle.current_group)
+				data["current_deathrattle_group"] = inserted_deathrattle.current_group
+	if(saved_deathrattle_group)
+		data["saved_deathrattle_group"] = saved_deathrattle_group.name
```

Comment:
Hardcoding subtype interactions makes me kinda sad

You don't have to change it if you don't want to but it would be really neat if we allowed implant subtypes to add/handle their own interactions which the UI just parses into something displayable

Reply:
that does sound really neat but I think that'd be a bit out of scope for this PR